public/src/components/alerts/Alert.tsx

```diff
@@ -0,0 +1,72 @@
+import { useEffect, useRef } from "react";
+import { Alert as AlertType } from "../../overmind/state";
+import { useActions } from "../../overmind";
+import React from "react";
+
+/** Alert is a component that displays a single alert.
+ * 
+ *  If the alert has a delay property, the alert will be removed after the delay.
+ *  In this case, an animated circle will be displayed to indicate the time remaining before the alert is removed.
+ * 
+ *  The alert is removed if the alert is clicked, or if the delay has passed.
+ *
+ *  @param alert - The alert to be displayed
+ */
+const Alert = ({ alert }: { alert: AlertType }): JSX.Element => {
+    const circleRef = useRef<SVGCircleElement>(null);
+    const actions = useActions();
+
+    useEffect(() => {
+        let id: ReturnType<typeof setTimeout>;
+        if (alert.delay) {
+            // If the optional delay property is set, the alert will be removed after the delay
```

Comment:
Maybe we don't need this comment here; since it is already documented in the top-level.